src/main/java/com/redhat/labs/lodestar/service/ComponentHealthService.java

```diff
@@ -0,0 +1,260 @@
+    private static final String DESIRED_REPLICAS = "Desired Replicas";
+    private static final String CONDITION_TYPE_AVAILABLE = "Available";
+
+    @Setter
+    @ConfigProperty(name = "component.namespaces", defaultValue = NOT_SET)
+    List<String> componentNamespaces;
+
+    @Setter
+    @ConfigProperty(name = "component.names", defaultValue = NOT_SET)
+    List<String> componentNames;
+
+    @Inject
+    OpenShiftApiClient client;
+
+    @Getter
+    Health health;
+
+    /**
+     * Initialize component names and namespaces lists if they were not configured.
+     * 
+     * @param event
+     */
+    void onStart(@Observes StartupEvent event) {
+
+        // initialize if namespaces not defined
+        if (componentNamespaces.size() == 1 && componentNamespaces.get(0).equals(NOT_SET)) {
+            componentNamespaces = new ArrayList<>();
+        }
+
+        LOGGER.debug("configured component namespaces {}", componentNamespaces);
+
+        // iniitalize if component names not set
+        if (componentNames.size() == 1 && componentNames.get(0).equals(NOT_SET)) {
+            componentNames = new ArrayList<>();
+        }
+
+        LOGGER.debug("Configured component names {}", componentNames);
+
+    }
+
+    /**
+     * Periodically reloads the {@link Health} for all components.
+     * 
+     */
+    @Scheduled(every = "10s")
+    void checkComponentHealth() {
+        getComponentHealth();
+    }
+
+    /**
+     * Creates a {@link Health} report using all {@link Check} found using the
+     * {@link OpenShiftApiClient}.
+     * 
+     */
+    public void getComponentHealth() {
+
+        // get checks
+        List<Check> checks = getChecks();
+
+        LOGGER.trace("found checks {}", checks);
+
+        // find overall status
+        String status = getOverallStatus(checks);
+
+        LOGGER.trace("determined overall status {}", status);
+
+        // set health to current state
+        health = Health.builder().status(status).checks(checks).build();
+
+    }
+
+    /**
+     * Returns a {@link List} of {@link Check} that summarize the status of
+     * {@link Deployment} and {@link DeploymentConfig} in the configured namespaces.
+     * 
+     * @return
+     */
+    private List<Check> getChecks() {
+
+        Stream<Check> deploymentChecks = getChecksForDeployments();
+        Stream<Check> deploymentConfigChecks = getChecksForDeploymentConfigs();
+
+        return Stream.concat(deploymentChecks, deploymentConfigChecks).collect(Collectors.toList());
+    }
+
+    /**
+     * Returns a {@link Stream} of {@link Check} for each {@link Deployment} in the
+     * configured namespaces.
+     * 
+     * @return
+     */
+    private Stream<Check> getChecksForDeployments() {
+
+        return componentNamespaces.stream()
+                .flatMap(namespace -> client.getDeploymentsByNamespace(namespace).stream())
+                .filter(deployment -> (componentNames.size() == 0
```

Comment:
should prob be componentNames. isEmpty()

Reply:
changed to isEmpty()